frontend/src/components/StakeSelectorItem.vue

```diff
@@ -1,217 +1,874 @@
       }
-      return exclamationMark;
     },
-    rewardTooltip() {
-      if(this.rewardDistributionTimeLeft > 0) {
-        if (this.currentRewardEarned > 0) {
-          return this.$t('stake.StakeSelectorItem.rewardsAvailableEarnedTooltip');
+
+    async onSubmit(): Promise<void> {
+      if (this.isLoadingStake || this.currentState !== CurrentState.ok) return;
+      const amount = this.bigNumberAmount.toString();
+
+      try {
+        this.isLoadingStake = true;
+
+        if (this.isDeposit) {
+          //stake
+          if(this.isNftStaking) {
+            await this.stakeNfts({ ids: this.idsToStake.map(x => x.id), stakeType: this.stakeType });
+          }
+          else {
+            await this.stake({ amount, stakeType: this.stakeType });
+          }
+        } else {
+          //unstake
+          if(this.stakeType === 'king') {
+            await this.unstakeKing({ amount });
+          }
+          else {
+            if(this.isNftStaking) {
+              await this.unstakeNfts({ ids: this.idsToStake.map(x => x.id), stakeType: this.stakeType });
+            }
+            else {
+              await this.unstake({ amount, stakeType: this.stakeType });
+            }
+          }
+        }
+        if(isNftStakeType(this.stakeType)) {
+          await this.updateOwnedLands();
+          this.idsToStake = [];
+        }
+      } catch (e) {
+        console.error(e);
+      } finally {
+        this.isLoadingStake = false;
+        this.textAmount = '';
+        this.idsToStake = [];
+      }
+    },
+    async onClaimReward(): Promise<void> {
+      if (this.rewardClaimState !== RewardClaimState.ok) return;
+
+      try {
+        this.isLoadingClaim = true;
+
+        if(this.stakeType === 'king') {
+          await this.claimKingReward();
         }
-        return this.$t('stake.StakeSelectorItem.rewardsAvailableTooltip');
+        else {
+          await this.claimReward({ stakeType: this.stakeType });
+        }
+      } catch (e) {
+        console.error(e);
+      } finally {
+        this.isLoadingClaim = false;
       }
-      if (this.currentRewardEarned > 0) {
-        return this.$t('stake.StakeSelectorItem.rewardsDepletedEarnedTooltip');
+    },
+    async fetchData(): Promise<void> {
+      try {
+        await this.fetchStakeDetails({ stakeType: this.stakeType });
+      } catch (err) {
+        console.log(err);
       }
-      return this.$t('stake.StakeSelectorItem.rewardsDepletedTooltip');
     },
+    async updateOwnedLands(): Promise<void> {
+      this.ownedLandIds = await this.getOwnedLandIdsWithTier();
+      this.stakedIds = await this.getStakedIds({ stakeType: this.stakeType });
+    }
   },
-};
+});
 </script>
 
 <style scoped>
 .container {
-  background: rgb(22, 22, 22);
-  background: linear-gradient(180deg, rgba(22, 22, 22, 1) 0%, rgba(16, 17, 17, 1) 100%);
-  padding: 1rem;
+  background: rgb(22, 22, 22); /* change to: background: #000E1D; */
```

Comment:
A TODO I suppose?

Reply:
Yes. When the whole new game ui is done we want to change the color theme.